pkg/services/ngalert/state/historian/annotation.go

```diff
@@ -61,11 +61,19 @@ func (h *AnnotationBackend) Record(ctx context.Context, rule history_model.RuleM
 	annotations := buildAnnotations(rule, states, logger)
 	panel := parsePanelKey(rule, logger)
 
+	// New background job, so start with a completely new context to avoid pollution.
+	writeCtx := context.Background()
+	writeCtx, cancel := context.WithTimeout(writeCtx, StateHistoryWriteTimeout)
+	writeCtx = history_model.WithRuleData(writeCtx, rule)
+
 	errCh := make(chan error, 1)
-	go func() {
+	go func(ctx context.Context) {
+		defer cancel()
 		defer close(errCh)
+		logger := h.log.FromContext(ctx)
```

Comment:
New logging context, therefore new logger. The new one ultimately contains the same fields as the old, but we keep things consistent this way, so there is less opportunity for future bugs. No problems due to closing over the wrong logger, etc.